# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/bindings/Codegen.py
# Commit: a6b021b3776a
# Full Hash: a6b021b3776ab954b0bf56d7a5641b072a719e2f
# Author: Kris Maglione <maglione.k@gmail.com>
# Date: 2020-06-27 09:38:32
# Regressor Bug: 1645510
# File Overlap Count: 1
# Description:
#   Bug 1645510: Part 3 - Make some UnprivilegedJunkScope calls fallible. r=mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D79721
# ==============================================================================

diff -r 93bca0225002 -r a6b021b3776a dom/bindings/Codegen.py
--- a/dom/bindings/Codegen.py	Sat Jun 27 03:06:28 2020 +0000
+++ b/dom/bindings/Codegen.py	Sat Jun 27 03:06:35 2020 +0000
@@ -14401,7 +14401,12 @@
                 // side-effects, followed by a call to JS::ToJSONMaybeSafely,
                 // which likewise guarantees no side-effects for the sorts of
                 // things we will pass it.
-                JSAutoRealm ar(cx, UnprivilegedJunkScopeOrWorkerGlobal());
+                JSObject* scope = UnprivilegedJunkScopeOrWorkerGlobal(fallible);
+                if (!scope) {
+                  JS_ReportOutOfMemory(cx);
+                  return false;
+                }
+                JSAutoRealm ar(cx, scope);
                 JS::Rooted<JS::Value> val(cx);
                 if (!ToObjectInternal(cx, &val)) {
                   return false;